src/bsoncxx/array/element.hpp

```diff
@@ -37,8 +37,6 @@ class BSONCXX_API element : private document::element {
    public:
     element();
 
-    explicit element(const std::uint8_t* raw, std::uint32_t length, std::uint32_t offset);
```

Comment:
This was unsafe to have as a public constructor, and not needed. I made it private and friended in the places that needed it. Noticed while looking into the coverity report about the uninit'd field in bson_iter_t.